src/strategies/citydao/index.ts

```diff
@@ -0,0 +1,46 @@
+import { Multicaller } from '../../utils';
+
+export const author = 'RosebuDAO';
+export const version = '0.1.0';
+
+const proofOfHumanityAbi = [
+  'function isRegistered(address _submissionID) external view returns (bool)'
+];
+const erc1155BalanceOfAbi = [
+  'function balanceOf(address owner, uint256 id) view returns (uint256)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  //verify humanity
+  const multi0 = new Multicaller(network, provider, proofOfHumanityAbi, { blockTag });
+  addresses.forEach((address) =>
+    multi0.call(address, options.proofOfHumanityContract, 'isRegistered', [address])
+  );
+  const result0: Record<string, number> = await multi0.execute();
+  //get balanceOf for humans
+  const multi1 = new Multicaller(network, provider, erc1155BalanceOfAbi, { blockTag });
+  const votes = new Map();
+  Object.entries(result0).forEach(([address, human]) => {
+    if (human) {
+      multi1.call(address, options.cityDaoCitizenNftContract, 'balanceOf', [address, options.cityDaoCitizenNftTokenId]);
+    }
+    else {
+      votes[address] = 0;
+    }
+  });
+  const result1: Record<string, number> = await multi1.execute();
+  Object.entries(result1).forEach(([address, balance]) => {
+    votes[address] = Math.min(balance, 1);
+  });
+  console.log(votes);
```

Comment:
Output of this console.log:
Map(0) {
      '0xf4d895311987faebd839908a0793461087b229c6': 0,
      '0x1db3439a222c519ab44bb1144fc28167b4fa6ee6': 1
    }


Really puzzled why this strategy isn't working at this point. The test suite fails on "Should return an array of object with addresses". It is seeing an empty object where it shouldn't.

Reply:
it's because I was using votes[address] = balance; .... instead of votes.set(address, balance);